java/arcs/core/data/proto/AccessPathProtoDecoder.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data.proto
+
+import arcs.core.data.AccessPath
+import arcs.core.data.HandleConnectionSpec
+
+/** Decodes an [AccessPathProto] into [AccesssPath]. */
+fun AccessPathProto.decode(
+    particleSpecName: String,
```

Comment:
Should this be encoded in the proto itself?

Reply:
That is a great idea! Had to make some changes on the typescript side. PTAL.

Reply:
LGTM!